### DISCUSSION TITLE
In an IVFPQ index, is it possible to get the distance of a query to a specified id vector?

### DISCUSSION THREAD

**Discussion by Learningisboundless** [facebookresearch/faiss#3666] [General]
Hi! Could the distance between the query vector and the specified id vector be quickly obtained in the IVFPQ index? If you use the reconstruct function and then use the pairwise_distances function to calculate, the time overhead will be relatively large

**Comment by mdouze** [comment]
It should not be that slow, did you enable the direct map ? 
https://github.com/facebookresearch/faiss/wiki/Special-operations-on-indexes#direct-map-for-an-indexivf

**Comment by Learningisboundless** [comment]
Very grateful for your help! In addition, I would like to ask if I know the vector id in the dataset, can I get its code and the invertedlist where it is located?
